dataactcore/models/jobModels.py

```diff
@@ -4,6 +4,17 @@
 from sqlalchemy.orm import relationship
 from dataactcore.models.baseModel import Base
 
+
+def generateFiscalYear(context):
+    """ Generate fiscal year based on the date provided """
+    reporting_end_date = context.current_parameters['reporting_end_date']
+    return reporting_end_date.year
```

Comment:
So this isn't quite enough, if the month is Oct-Dec, you need to add one to the year to bring it into the correct fiscal year.